tss-esapi/src/structures/buffers.rs

```diff
@@ -23,8 +23,24 @@ macro_rules! named_field_buffer_type {
         impl $native_type {
             pub const MAX_SIZE: usize = $MAX;
 
-            pub fn value(&self) -> &[u8] {
-                &self.0
+            pub fn from_bytes(bytes: &[u8]) -> Result<Self> {
+                Self::ensure_valid_buffer_size(bytes.len(), "bytes(&[u8])")?;
+                Ok($native_type(bytes.to_vec().into()))
+            }
+
+            /// Returns the content of the buffer type as
+            /// a slice of bytes.
+            pub fn as_bytes(&self) -> &[u8] {
```

Comment:
Would it make sense to have a `From<$native_type> for Vec<u8>` implemented? It would avoid a lot of `.as_bytes().to_vec()`

Reply:
Though it is a little bit risky to do stuff like that because then that vector wont be zeroized when dropped and I wonder if the user really wants that and we do implement ```deref```.

Reply:
I mean, the user becomes responsible for the data - we already offer access to it, they can clone it and do the same things, this would just make it easier. We can perhaps put a warning on the API about it.